portal-kernel/src/com/liferay/asset/kernel/service/persistence/AssetEntryQuery.java

```diff
@@ -150,7 +150,11 @@ public AssetEntryQuery(
 
 		if (Validator.isNotNull(tagName)) {
 			_allTagIds = AssetTagLocalServiceUtil.getTagIds(
-				themeDisplay.getSiteGroupId(), new String[] {tagName});
+				themeDisplay.getScopeGroupId(), new String[] {tagName});
+
+			if (_allTagIds.length == 0) {
```

Comment:
Got it, thanks. Go ahead and address the other two concerns and send a pull with the new fix when it's ready.